File: Problem-028/solution.js
function spiralDiagonals(n) {
  let sum = 1;
  for( let size = 3; size<= n; size +=2){
    let square = size*size;
    let step = size - 1;
    sum += square;
    sum += square - step;
    sum += square - 2 * step;
    sum += square - 3 * step;
  }
  return sum;
}

File: Problem-028/solution.py
def number_spiral_diagonals(n):

    total = 1

    for size in range(3, n + 1, 2):

        square = size * size
        step = size - 1

        total += square
        total += square - step
        total += square - 2 * step
        total += square - 3 * step

    return total
